### PR TITLE
Add NATS consumer input plugin.

### PR DESCRIPTION
**Pull request by netixen**
Used the kafka consumer as a blueprint.
More info about [NATS](http://www.nats.io/about/)

### CODE REVIEW

File: Godeps [hunk 1]
```diff
@@ -20,17 +18,17 @@ github.com/gorilla/mux 26a6070f849969ba72b72256e9f14cf519751690
 github.com/hailocab/go-hostpool e80d13ce29ede4452c43dea11e79b9bc8a15b478
 github.com/influxdata/config bae7cb98197d842374d3b8403905924094930f24
 github.com/influxdata/influxdb a9552fdd91361819a792f337e5d9998859732a67
+github.com/influxdata/telegraf 0109320e226130940cd0b999ea529c05cda49a88
```
Comment (netixen) [inline Godeps]: I guess 'gdm save' shouldn't do this?

**Comment by pires** [comment]
@netixen even though I can't validate the PR with testing, I did a quick read and congratulations for the amazing job! I know a lot of NATS users will appreciate this.

**Comment by bjflanne** [comment]
@netixen second what @pires said! - FYI I am NATS Community Manager. If quick chat with anyone on our team would be useful to you in working on this PR just reach out - we can schedule something, or if you are a Slack user tell me and I will get you in the Slack Community for NATS..

brian@apcera.com

B

**Comment by netixen** [comment]
@pires @bjflanne Thank You. I'm in love with the elegance and speed of NATS. And since influxdb is designed for hundreds of thousands of writes per second, it's a great match. Awesome job from apcera and InfluxData..

**Comment by bjflanne** [comment]
Excellent! Yes -- simplicity and performance are two core philosophies NATS was designed around. Where are you located? There's a NATS meetup in San Francisco next month if you'd like to give lightning talk on your plugin happy to give you some time on the agenda. Likewise, if you're a slack user pop me an email and I can get you into the Slack community. Keep up the good work!!!!

B

File: plugins/inputs/nats_consumer/nats_consumer.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  ### urls of NATS servers
+  servers = ["nats://localhost:4222"]
+  ### Use Transport Layer Security
+  secure = false
+  ### subject(s) to consume
+  subjects = ["telegraf"]
+  ### name a queue group
+  queue_group = "telegraf_consumers"
+  ### Maximum number of points to buffer between collection intervals
+  point_buffer = 100000
```
Comment (sparrc) [inline plugins/inputs/nats_consumer/nats_consumer.go]: can you change `point_buffer` to `metric_buffer`? I have a PR making this change in the kafka consumer already

Comment (sparrc) [inline plugins/inputs/nats_consumer/nats_consumer.go]: see #676

Comment (netixen) [inline plugins/inputs/nats_consumer/nats_consumer.go]: @sparrc Done

**Comment by sparrc** [comment]
thanks @netixen, looks good

**Pull request closed without merging**